addons/point_of_sale/static/src/app/store/db.js

```diff
@@ -567,22 +613,29 @@ export class PosDB {
         } catch {
             return [];
         }
-        var results = [];
-        for (var i = 0; i < this.limit; i++) {
-            var r = re.exec(this.category_search_string[category_id]);
-            if (r) {
-                var id = Number(r[1]);
-                const product = this.get_product_by_id(id);
-                if (!this.shouldAddProduct(product, results)) {
+        let results = [];
+        if(filteredProducts.length > 0) {
+            for(const filteredProduct of filteredProducts){
```

Comment:
```suggestion
            for (const filteredProduct of filteredProducts) {
```
(same for the if below and above)